### PR TITLE
New package: typespeed-0.6.5

### PR DESCRIPTION
**Pull request by kalichakra**

### CODE REVIEW

File: srcpkgs/typespeed/template [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Template file for 'typespeed'
+pkgname="typespeed"
+version="0.6.5"
+revision=1
+build_style=gnu-configure
+makedepends="ncurses-devel"
+short_desc="Test your typing speed"
+only_for_archs="x86_64 i686 aarch64 armv7hf"
```
Comment (Vaelatern) [inline srcpkgs/typespeed/template]: Please don't do this. For example, you have just unintentionally banned `armv6hf`, for which this probably builds just fine. Remove this line, let someone else decide how to fix `-musl`, or if to do so at all.

File: srcpkgs/typespeed/template [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+maintainer="David <kalichakra@zoho.com>"
+license="GPL-2"
+homepage="http://typespeed.sourceforge.net"
+distfiles="${homepage}/${pkgname}-${version}.tar.gz"
```
Comment (Vaelatern) [inline srcpkgs/typespeed/template]: We have a more "official" way to download from sourceforge, documented in the Manual. Please use it.

File: srcpkgs/typespeed/template [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+checksum="5c860385ceed8a60f13217cc0192c4c2b4705c3e80f9866f7d72ff306eb72961"
+nocross=yes
```
Comment (Vaelatern) [inline srcpkgs/typespeed/template]: Please remove this line too.

**Comment by kalichakra** [comment]
Fixed

**Comment by Vaelatern** [comment]
https://travis-ci.org/voidlinux/void-packages/jobs/183317693#L644
Seems like clock_t is defined in muslc, have to patch to include that header file in the right place. (I might suggest a patch later)

**Comment by Vaelatern** [comment]
https://travis-ci.org/voidlinux/void-packages/jobs/183317692#L791-L794
It's calling `gcc` instead of `$(CC)`, which will cause problems. This should also be patched.

**Comment by kalichakra** [comment]
I've looked into clock_t and muslc, indeed it's defined in "bits/alltypes.h", and "time.h" would need to be included before as well, and these should be included from "src/typespeed.h". Don't waste your time on this i'll provide a patch.

As for gcc, src/Makefile.am uses "CC = gcc", which could just be commented i suppose ?

**Comment by kalichakra** [comment]
Updated. Fixed compilation on -musl arches. CC=gcc is stripped from the main makefiles.

**Pull request merged**